clientbase/common.go

```diff
@@ -183,6 +184,17 @@ func NewAPIClient(opts *ClientOpts) (APIBaseClient, error) {
 
 	client.Timeout = opts.Timeout
 
+	var proxy func(*http.Request) (*url.URL, error)
+	if opts.ProxyURL != "" {
+		proxyURL, err := url.Parse(opts.ProxyURL)
+		if err != nil {
+			return result, err
```

Comment:
Nit:

If we can wrap this error, that would provide more context on what specific part of the call failed. Something like:

```suggestion
			return result, fmt.Errorf("unable to parse url from url specified in options: %w", err)
```

This change isn't strictly necessary, but might help a bit with longer-term usability.

Reply:
Good point, returning the error now in the same way as `ProxyFromEnvironment` [does](https://github.com/golang/net/blob/master/http/httpproxy/proxy.go#L164) if it fails to parse the url.